pennylane/resource/error.py

```diff
@@ -77,3 +78,43 @@ def error(self) -> AlgorithmicError:
         Returns:
             AlgorithmicError: The error.
         """
+
+
+class SpectralNormError(AlgorithmicError):
+    """Class representing the spectral norm error.
+
+    The spectral norm error is defined as the distance (in spectral norm) between the true unitary
+    we wish to apply and the approximate unitary that we actually apply.
+
+    Args:
+        error (float): The numerical value of the error
+
+    """
+
+    def combine(self, other: "SpectralNormError"):
+        """A method to combine two spectral norm errors.
+
+        Args:
+            other (SpectralNormError): The other instance of error being combined.
+
+        Returns:
+            SpectralNormError: The total error after combination.
+        """
+        return self.__class__(self.error + other.error)
+
+    @staticmethod
+    def get_error(approximate_op: Operator, exact_op: Operator, **kwargs):
+        """Compute spectral norm error between two operators.
+
+        Args:
+            approximate_op (Operator): The approximate operator.
+            exact_op (Operator): The exact operator.
+
+        Returns:
+            float: The error between the exact operator and its
+            approximation.
+        """
+        if approximate_op.has_matrix and exact_op.has_matrix:
+            return fn.svd(exact_op.matrix() - approximate_op.matrix(), compute_uv=False)[0]
```

Comment:
We should make sure the wire order is the same between the two matrices, so I would say something like: 

```
wire_order = exact_op.wires
return fn.svd(exact_op.matrix() - approximate_op.matrix(wire_order=wire_order), ...)
```